packages/rollup-core/src/app/data/data-service.ts

```diff
@@ -452,6 +457,43 @@ export class DefaultDataService implements DataService {
     }
   }
 
+  public async getMaxL2TxOutputIdForCanonicalChainBatch(
+    batchTimestamp: number,
+    maxBatchCalldataBytes: number
+  ): Promise<number> {
+    const res: Row[] = await this.rdb.select(
+      `SELECT id, (LENGTH(calldata)-2) / 2 as calldata_bytes
```

Comment:
Same thing about a bug if `LENGTH(calldata)` < 2

Reply:
Good call! Will make it `MAX(LENGTH(calldata)-2, 0) / 2`